### PR TITLE
Refine code(momentum_op, rmsprop_op )

### PR DESCRIPTION
**Pull request by chengduoZH**
fix https://github.com/PaddlePaddle/Paddle/issues/6002

### CODE REVIEW

File: paddle/operators/momentum_op.h [hunk 1]
```diff
@@ -50,8 +50,7 @@ class MomentumOpKernel : public framework::OpKernel<T> {
 
     v_out.device(place) = v * mu + g;
     if (use_nesterov) {
-      p_out.device(place) = p - g * lr.broadcast(grad_dsize) +
-                            v_out * mu * lr.broadcast(grad_dsize);
+      p_out.device(place) = p - (g - v_out * mu) * lr.broadcast(grad_dsize);
```
Comment (qingqing01) [inline paddle/operators/momentum_op.h]: Whether the `framework::EigenScalar` can be used for `lr` instead of `broadcast` ?

Comment (dzhwinter) [inline paddle/operators/momentum_op.h]: This implement is almost same.
it's better remove the `v_out.device`

Comment (chengduoZH) [reply paddle/operators/momentum_op.h]: I've tried it, but the result is wrong.

Comment (chengduoZH) [reply paddle/operators/momentum_op.h]: Great, I also think it's better to remove the `v_out.device`

**Comment by chengduoZH** [comment]
I find that the implementation of momentum_op seems to be  wrong.
[Document description](http://www.paddlepaddle.org/docs/develop/documentation/zh/operators.html#permalink-149-momentum):
``` 
velocity = mu * velocity + gradient 
if (use\_nesterov):   
  param = param - gradient * learning\_rate + mu * velocity * learning\_rate 
else:   
  param = param - learning\_rate * velocity. 
```
Code implementation:
https://github.com/PaddlePaddle/Paddle/blob/728e8b1708abb683a4dd65a6a3459b4dfa146eff/paddle/operators/momentum_op.h#L51-L57

File: paddle/operators/momentum_op.h (hunk 1 above)
Comment (chengduoZH) [reply paddle/operators/momentum_op.h]: I find that `v_out` and `v` point to the same memory, this is to say the writer use inplace method in python interface. And it is necessary to update `v_out` in momentum_op, so `v_out.device` can't be removed.

**Comment by chengduoZH** [comment]
The writer use inplace method in python interface, this is to say that  `v_out` and `v` point to the same memory. I think this should be annotated in the code.
https://github.com/PaddlePaddle/Paddle/blob/21053c168cd842565b015271a17d0e2aa116e40e/python/paddle/v2/fluid/optimizer.py#L265-L278

**Pull request closed without merging**